libvast/vast/concept/convertible/data.hpp

```diff
@@ -396,42 +396,33 @@ class record_inspector {
     if (*data_value == nullptr)
       return caf::none;
     auto err = caf::visit(
-      [&]<class Type>(const Type& t) -> caf::error {
+      [&]<class Data, class Type>(const Data& d, const Type& t) -> caf::error {
         using concrete_type = std::decay_t<Type>;
-        using concrete_data
-          = std::conditional_t<std::is_same_v<concrete_type, enumeration_type>,
-                               std::string, type_to_data<concrete_type>>;
         if constexpr (detail::is_any_v<concrete_type, alias_type, none_type>) {
           // Data conversion of none or alias type does not make sense.
```

Comment:
Isn't recursing with the inner alias type what we want?

Reply:
I'm not sure in which situation you would use an alias type in the definition of the layout of a native type?

Reply:
Hmm, semantics? I don't have a concrete use case at hand, sorry. Maybe we'll just wait until/if this pops up.